### PR TITLE
UnitTest enchancement

### PR DESCRIPTION
**Pull request by babenek**
Separate application tests and unit-tests
Skip additional filter for UT

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -59,3 +59,11 @@ def scanner(rule: Rule, config: Config, rule_path: str) -> Scanner:
     scanner = Scanner(config, rule_path)
     scanner.rules = [rule]
     return scanner
+
+
+@pytest.fixture
+def scanner_without_filters(rule: Rule, config: Config, rule_path: str):
```
Comment (meanrin) [inline tests/conftest.py:65]: Some tests do use randomly seeded credentials, so they have a small change to be filtered out by a pattern filter (for example by randomly generating `aaaa` sub-string
This `scanner_without_filters` fixture allows to run regex related tests without filters that can remove some random creds

All other tests still using old `scanner` fixture

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -41,7 +41,10 @@ jobs:
         pip install .[ml]
     - name: Test with pytest
       run: |
-        python -m pytest --random-order --random-order-bucket=global --ignore=docs -m "not api_validation" tests/
+        # unit-tests
+        python -m pytest --random-order --random-order-bucket=global --ignore=docs --ignore=tests/test_app.py tests/
+        # application test
+        python -m pytest --random-order --random-order-bucket=global --ignore=docs tests/test_app.py
```
Comment (meanrin) [inline .github/workflows/test.yml:47]: I assume that new test command can be separated to a new step in the `test.yml`
To add clearer divide between unit and integration tests, if we want to 

Btw, I don't see direct benefits dividing tests into 2 separate commands, they are still both just a pytest tests anyway

Comment (meanrin) [inline .github/workflows/test.yml:47]: I assume that new test command can be separated to a new step in the `test.yml`

Comment (babenek) [reply .github/workflows/test.yml:47]: Unit tests might be used for code coverage. Separating steps lead to overhead in workflow - double python setup.

Comment (meanrin) [reply .github/workflows/test.yml:47]: I meant something like

```yaml
- name: Unit tests
      run: ...
- name: Application tests
      run: ... 
```

That should not add any overhead, just additional separation between different types of tests 

image

File: tests/conftest.py (hunk 1 above)
Comment (babenek) [reply tests/conftest.py:65]: UnitTests should avoid any "chance of".

File: .github/workflows/test.yml (hunk 2 above)
Comment (babenek) [reply .github/workflows/test.yml:47]: ok

**Review by kmnls — APPROVED** [review]

**Review by iuriimet — APPROVED** [review]
Agree

**Pull request merged**